### PR TITLE
Add initial ROS2 humble packages

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
Add initial ROS2 humble packages

NOTE: CI is modified for now to read from a list and build. In the future, only packages exist in this repository will be built.
These are packages already building fine in this PR: https://github.com/usdot-fhwa-stol/carma-platform/pull/2472

NOTE: There seem a lot of changes, but they are all basically following changes:
- needed to include tf2_buffer in gnss_to_map_converter instead of using #include <tf2/LinearMath/Vector3.h>
- declare_parameter needed default values
- changing from header .h to .hpp
- using rclcpp::Duration::from_nanoseconds(<int_nanoseconds>) function as the rclcpp::Duration(<int_nanoseconds>) constructor no longer works
- tf2_listener and tf2_buffer needed to be made into shared_ptr

## Related GitHub Issue
NA

## Related Jira Key
ARC-233

## Motivation and Context
ROS2 Humble Upgrade

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by MishkaMN** [comment]
Currently CI is failing only because of lint checks in the unit tests.

File: .sonarqube/sonar-scanner.properties [hunk 1]
```diff
@@ -46,7 +46,6 @@ sonar.modules= bsm_generator, \
   carma_wm, \
   carma_wm_ctrl, \
   roadway_objects, \
-  platooning_strategic_IHP, \
```
Comment (john-chrosniak) [inline .sonarqube/sonar-scanner.properties:49]: Should this be replaced with platooning_strategic_ihp?

File: carma/launch/plugins.launch.py [hunk 2]
```diff
@@ -559,12 +559,12 @@ def generate_launch_description():
         namespace=GetCurrentNamespace(),
         composable_node_descriptions=[
             ComposableNode(
-                package='platoon_strategic_ihp',
-                plugin='platoon_strategic_ihp::Node',
-                name='platoon_strategic_ihp_node',
+                package='platooning_strategic_ihp',
```
Comment (john-chrosniak) [inline carma/launch/plugins.launch.py:562]: These changes should also be made in the package.xml and CMakeLists.txt of `platooning_strategic_ihp`

File: docker/install.sh [hunk 3]
```diff
@@ -40,30 +27,14 @@ sudo mkdir -p /opt/carma # Create install directory
...
-fi
-echo "Build of ROS1 CARMA Components Complete"
+# Source the autoware installation
 
-###
-# ROS2 installation
-###
-# Source the ROS2 autoware installation
-source /home/carma/catkin/setup.bash
 if [[ ! -z "$ROS1_PACKAGES$ROS2_PACKAGES" ]]; then
```
Comment (john-chrosniak) [inline docker/install.sh:32]: Can probably remove the `ROS1_PACKAGES` variable

File: subsystem_controllers/config/guidance_controller_config.yaml [hunk 4]
```diff
@@ -51,6 +51,6 @@
       - /guidance/plugins/stop_and_wait_plugin
       - /guidance/plugins/sci_strategic_plugin
       - /guidance/plugins/stop_controlled_intersection_tactical_plugin
-      - /guidance/plugins/platoon_strategic_ihp
+      - /guidance/plugins/platooning_strategic_ihp
```
Comment (john-chrosniak) [inline subsystem_controllers/config/guidance_controller_config.yaml:54]: Comment not specific for here but there are still references to `platoon_strategic_ihp`

**Review by john-chrosniak — CHANGES_REQUESTED** [review]
Confirmed everything builds in ROS2 Humble

File: .github/workflows/Ci.yml [hunk 5]
```diff
@@ -63,64 +63,36 @@ jobs:
         source "$INIT_ENV"
         ./src/${{ github.event.repository.name }}/docker/checkout.bash -r /opt/carma/ -b ${{ steps.determine-base-branch.outputs.git_branch }}
 
-    - name: Install external dependencies
-      # Install the multiple object tracking deps
-      run: sudo bash /opt/carma/src/multiple_object_tracking/scripts/install_dependencies.sh
```
[the remark this answers is not in the document]
Comment (MishkaMN) [reply .github/workflows/Ci.yml:68]: we should keep this, I just had it turned off, but let me turn them on.

File: .sonarqube/sonar-scanner.properties (hunk 1 above)
Comment (MishkaMN) [reply .sonarqube/sonar-scanner.properties:49]: goto catch, added

File: carma/launch/plugins.launch.py (hunk 2 above)
Comment (MishkaMN) [reply carma/launch/plugins.launch.py:562]: Anish is handling this in his PR, so we can probably leave this for now

File: docker/install.sh (hunk 3 above)
Comment (MishkaMN) [reply docker/install.sh:32]: removed

File: subsystem_controllers/config/guidance_controller_config.yaml (hunk 4 above)
Comment (MishkaMN) [reply subsystem_controllers/config/guidance_controller_config.yaml]: I actually just reverted the chagnes for this plugin and let Anish handle it fully in his PR

**Comment by MishkaMN** [comment]
Ignoring SonarScan failure because there is a separate stories as it is not part of the scope to add more unit tests for Humble upgrade.

File: lightbar_manager/include/lightbar_manager/lightbar_manager_node.hpp [hunk 6]
```diff
@@ -47,9 +47,9 @@ struct Config
 {
     double spin_rate_hz = 10.0;
     bool normal_operation = true;   // if false, other plugins are able to take control over the lightbar status
-    std::vector<std::string> lightbar_priorities = {}; // Tsesters are for unit testing. Keep it there.
-    std::vector<std::string> lightbar_cda_table = {}; // Keys for lightbar_cda_to_ind_table, 1-to-1 with lightbar_ind_table
-    std::vector<std::string> lightbar_ind_table = {}; // Values for lightbar_cda_to_ind_table, 1-to-1 with lightbar_cda_table
+    std::vector<std::string> lightbar_priorities = {"lightbar_manager"}; // Tsesters are for unit testing. Keep it there.
```
[the remark this answers is not in the document]
Comment (MishkaMN) [reply lightbar_manager/include/lightbar_manager/lightbar_manager_node.hpp]: I removed the comment anyways because it was wrong.

**Review by john-chrosniak — APPROVED** [review]

**Pull request merged**